igb/repl.go

```diff
@@ -377,6 +507,40 @@ func newIVM() (ivm iVM, err error) {
 	return ivm, nil
 }
 
+// Returns true if double quotation in the string is open.
+func checkDoubleQuoteOpen(s string) bool {
+	openDoubleQuote, _ := regexp2.Compile(`^[^']*"`, 0)
+
+	s = strings.Replace(s, "\\\"", "", -1)
+	s = strings.Replace(s, "\\'", "", -1)
+	od, _ := openDoubleQuote.MatchString(s)
+	if strings.Count(s, "\"")%2 == 1 && od {
+		return true
+	}
+	return false
+}
+
+// Returns true if single quotation in the string is open.
+func checkSingleQuoteOpen(s string) bool {
+	openSingleQuote, _ := regexp2.Compile(`^[^"]*'`, 0)
+
+	s = strings.Replace(s, "\\\"", "", -1)
+	s = strings.Replace(s, "\\'", "", -1)
+	sd, _ := openSingleQuote.MatchString(s)
+	if strings.Count(s, "'")%2 == 1 && sd {
+		return true
+	}
+	return false
+}
+
+// prompt switches prompt sign.
+func prompt(s int) string {
```

Comment:
I think we should name this as an action, like `switchPrompt`

Reply:
You're correct. Updated.